Code-For-Chicago/Voma
# Added user authentication.

Added user authentication and hooked up the login form.

Push requires a resync of your database table structure and `npm install` on both server and client.

- Changed models to use 'underscores' for created_at and updated_at. Need these like this for the lando sync script.
- (Lando Config) Modified the database bash script to seed test data for Projects, Skills, Volunteers, Volunteer Skills. Added associations. Left a few volunteers without an associated project. 
- Pluralized "skill" on FE and BE hook.
- Modified the Lando sync script to push a user into the admin table. Credentials are voma.code.for.chicago@gmail.com with the password "password".
- Added express-session plugin and configured it for development.
- Refactored the login function in the admins controller to work with the login form. Just authentication is set atm we can add user type when those are figured out and ready.
- Stubbed out loginState function, this will get hit on page load and force a logout for unauthenticated users and reset the max-age of the session cookie.
- Added login function to the AuthProvider. On success redirect to `/board`
- Added error messaging to the Login form.
- Locked routes in Express that are not public.
- Setup proxy middleware in the client so that cookies from port 5000's server would set on the port 3000 frontend.

**Testing with Lando config:**
- Pull down, run `npm install` then wipe your box and rebuild it.
- `lando destroy` then `lando start` then `bash .lando.database.sh`
- Turn on the server and client. The credentials for an admin user should have been added by the database bash script.  Credentials are voma.code.for.chicago@gmail.com with the password "password". 

**Testing with other Databse Setup:**
- Pull down, run `npm install` then start the client and server.
- Add and admin user to the table, this is the correct hash for the password "password" .. `$2b$10$tNLcr6nHUm30/JEquBjCbuii77MHx4qXqjUc8JRRx0MnSW6FXT.BK`
